kernel/arch/amd64/include/arch/cpuid.h

```diff
@@ -56,7 +56,7 @@ typedef struct {
 	uint32_t cpuid_edx;
 } __attribute__ ((packed)) cpu_info_t;
 
-extern int has_cpuid(void);
+extern errno_t has_cpuid(void);
```

Comment:
This is rather a true/false or no bits set / some bits set variable, there is no error information in it.

Reply:
Maybe an unsigned type would be best here.

Reply:
Maybe, but that's out of scope of this changeset. :)